### ISSUE TITLE
Question about reconstruction loss in vqvae.py

### ISSUE THREAD

**Issue by Tigerrr07** [junxia97/Mole-BERT#17]
Thank you for your excellent work! I'm interested in the way you use VQ-VAE method and I have read your paper carefully. However, I'm confused about Equation 4 in your paper which includes reconstruction loss between input attributes and reconstructed attributes, but this loss doesn't seem to appear for the following code:
https://github.com/junxia97/Mole-BERT/blob/2feff8a33e3634b66b7408e2e2780fc9d960909f/vqvae.py#L42-L52
Could you please help me understand why this is the case or I missed something in your paper? I appreciate your time and attention.

**Comment by junxia97** [comment]
https://github.com/junxia97/Mole-BERT/blob/2feff8a33e3634b66b7408e2e2780fc9d960909f/vqvae.py#L148

Here is the reconstruction loss.

**Comment by Tigerrr07** [comment]
> https://github.com/junxia97/Mole-BERT/blob/2feff8a33e3634b66b7408e2e2780fc9d960909f/vqvae.py#L148
> 
> Here is the reconstruction loss.

Thank you for your reply! I mistook it that tokenizer was pre-training together with MAM. But I have some questions about the codebook.
The tokenizer is `DiscreateGNN` defined in model.py.
https://github.com/junxia97/Mole-BERT/blob/2feff8a33e3634b66b7408e2e2780fc9d960909f/model.py#L423-L429

In pretrain.py, the code shown below calls `get_codebook_indices` method to get token ids. But `self(*argv)` will only get [N, E] node representations instead of `logits` in the `get_codebook_indices` method (I think).
https://github.com/junxia97/Mole-BERT/blob/2feff8a33e3634b66b7408e2e2780fc9d960909f/pretrain.py#L143

I also notice that the codebook was defined but not was used.
https://github.com/junxia97/Mole-BERT/blob/2feff8a33e3634b66b7408e2e2780fc9d960909f/pretrain.py#L244-L245

My questions are:
* Why is the codebook defined but not used in pretrain.py? Is this a bug or a feature?
* How should I modify the pretrain.py file to use the codebook? Should I change something for line 143 like this? 
  ``` python
  node_rep = tokenizer(batch_origin_x, batch1.edge_index, batch_origin_edge)
  atom_ids = codebook.get_code_indices(node_rep)
  ```
    Atom representations are first obtained by tokenizer, and codebook was used to find the nearest token for every atom.

I look forward to hearing from you. Thank you for your support and contribution.
